packages/authenticator/amplify_authenticator/lib/src/blocs/auth/auth_bloc.dart

```diff
@@ -333,6 +339,43 @@ class StateMachineBloc
             allowedMfaTypes: result.nextStep.allowedMfaTypes,
           ),
         );
+      case AuthSignInStep.continueSignInWithMfaSetupSelection:
+        final allowedMfaTypes = result.nextStep.allowedMfaTypes;
+        if (allowedMfaTypes != null) {
+          final mfaTypesForSetup = allowedMfaTypes.toSet()..remove(MfaType.sms);
+          if (mfaTypesForSetup.length == 1) {
+            final mfaType = mfaTypesForSetup.first;
+            if (mfaType == MfaType.totp) {
+              assert(
+                result.nextStep.totpSetupDetails != null,
+                'Sign In Result should have totpSetupDetails',
+              );
+              _emit(await ContinueSignInTotpSetup.setupURI(
+                result.nextStep.totpSetupDetails!,
+                totpOptions,
+              ),);
+            } else if (mfaType == MfaType.email) {
+              _emit(UnauthenticatedState.continueSignInWithEmailMfaSetup);
+            } else {
+              throw InvalidUserPoolConfigurationException(
+                'Unsupported MFA type: ${mfaType.name}',
+                recoverySuggestion: 'Check your user pool MFA configuration.',
+              );
+            }
+          } else {
+            _emit(
+              ContinueSignInWithMfaSetupSelection(
+                allowedMfaTypes: result.nextStep.allowedMfaTypes,
+              ),
+            );
+          }
+        } else {
+          _emit(
+            ContinueSignInWithMfaSetupSelection(
+              allowedMfaTypes: result.nextStep.allowedMfaTypes,
+            ),
+          );
```

Comment:
Q: this seems that if `AuthSignInStep.continueSignInWithMfaSetupSelection` and `result.nextStep.allowedMfaTypes == null` then it
`_emit(ContinueSignInWithMfaSetupSelection(
              allowedMfaTypes: result.nextStep.allowedMfaTypes,
            ),);` isn't it causing the state machine to stuck on `AuthSignInStep.continueSignInWithMfaSetupSelection`?